apisix/cli/ngx_tpl.lua

```diff
@@ -432,6 +436,11 @@ http {
             dns_resolver = dns_resolver,
         }
         apisix.http_init(args)
+
+        {% if enabled_plugins["saml-auth"] then %}
+        local saml = require("apisix.plugins.saml-auth")
```

Comment:
We should not require a plugin here. This will make it unable to disable the plugin.

Reply:
So where do you prefer to set the data_dir? Define `set_data_dir()` in lua-resty-saml and call it here? The `apisix.constants` is not suitable according to @membphis.

Reply:
we can do this in `init` phase

Reply:
@membphis No matter which place to init the plugin, we need to determine `apisix_home` before apisix start, because saml init needs the absolute path of `xsd/` dir.

Reply:
What about putting the apisix_lua_home in the constants module, so we can use it in the plugin?

Reply:
That's ok.